cl/corpus_importer/tests.py

```diff
@@ -2387,3 +2402,497 @@ def test_find_and_fix_mis_matched_dockets(self, mock_download_ia):
         # The mis matched docket is preserved, fix its source to RECAP.
         mis_matched_docket.refresh_from_db()
         self.assertEqual(mis_matched_docket.source, Docket.RECAP)
+
+
+class HarvardMergerTests(TestCase):
+    def setUp(self):
+        """Setup harvard merger tests"""
+        self.read_json_patch = patch(
+            "cl.corpus_importer.management.commands.harvard_merge.read_json"
+        )
+        self.read_json_func = self.read_json_patch.start()
```

Comment:
Huh, I haven't seen this before. How does this approach differ from the decorator approach of patching?

Reply:
Good question.  Its ben a while let me check

Reply:
The approach is almost the same as if you were using the decorator, the difference is that we pass all the arguments to the decorator and it automatically returns the created mock, and using this approach we need to specify the function to mock, then activate the patch and specify the return value, all done in individual steps.